### ISSUE TITLE
Print fails to display document preview intermittently with iOS 17 upgrade

### ISSUE THREAD

**Issue by shekhartupe** [katzer/cordova-plugin-printer#301]
We are using this plugin in an ionic app. With iOS 17 upgrade, we see this behavior where print menu fails to display the preview intermittently. Attached screenshot of the behavior and console logs from app.

![Screenshot 2023-10-21 at 4 19 28 PM](https://github.com/katzer/cordova-plugin-printer/assets/12859778/abe9525f-af86-45ab-90f9-048df97fef65)

[preview-failed.txt](https://github.com/katzer/cordova-plugin-printer/files/13061773/preview-failed.txt)
[preview-success.txt](https://github.com/katzer/cordova-plugin-printer/files/13061774/preview-success.txt)

Any guidance will help.

**Comment by shekhartupe** [comment]
Just to add, if I retry printing same file again and again, sometimes it will display the preview.

**Comment by shekhartupe** [comment]
Anyone else facing similar issue or seen this in past?

**Comment by kirtipriya** [comment]
My iPad app supports only landscape orientation. The moment print button is clicked in the app , the app crashes with the error :
*** Terminating app due to uncaught exception 'UIApplicationInvalidInterfaceOrientation', reason: 'Supported orientations has no common orientation with the application, and [UIPrintPanelNavigationController shouldAutorotate] is returning YES'

**Comment by davequested** [comment]
> Anyone else facing similar issue or seen this in past?

@shekhartupe We're seeing exactly the same problem. Did you get to the bottom of it?

**Comment by shekhartupe** [comment]
@davequested yes, in my case I am writing a file locally and then printing the same. I had file check api to ensure file was created and then only trigger print api. This worked well before iOS 17.x. With 17.x I observed even though file was created, it was not released to open and hence I added a delay of 1 sec after file check to trigger print api. This workaround resolved issue for me with this plugin.

**Comment by davequested** [comment]
Unfortunately this doesn't make any difference for me. We just sent the
print() method raw html. Works fine on Android and iOS16. Apple have
changed something. It's linked to third party assets. If we don't have any
resources (e.g. css, images) from a third party URL then it works fine.
Perhaps they've got some permission hoops we now need to jump through?
